### PR TITLE
Login verification supports jwt mode

### PR DESCRIPTION
**Pull request by elonlo**
## What is the purpose of the pull request

*Refactor login authentication process, use interface and add jwt authentication mode*

## Brief change log

  - *Add io.jsonwebtoken.jjwt to root pom.xml*
  - *Add io.jsonwebtoken.jjwt to dolphinscheduler-api pom.xml*
  - *Add AuthenticationType file to dolphinscheduler-api module*
  - *Add Authenticator file to dolphinscheduler-api module*
  - *Add JsonWebTokenAuthenticator file to dolphinscheduler-api module*
  - *Add PasswordAuthenticator file to dolphinscheduler-api module*
  - *Add SecurityConfig file to dolphinscheduler-api module*
  - *Add JsonWebTokenAuthenticatorTest file to dolphinscheduler-api module*
  - *Add PasswordAuthenticatorTest file to dolphinscheduler-api module*
  - *Add SecurityConfigTest file to dolphinscheduler-api module*
  - *Modify LoginController file*
  - *Modify TenantService file*
  - *Modify UsersService file*

## Verify this pull request

  - *Added PasswordAuthenticatorTest to verify password authenticator.*
  - *Added JsonWebTokenAuthenticatorTest to verify rsa-public-key.*
  - *Add SecurityConfigTest to verify authenticator is available.*
  - *Manually verified the change by testing locally.*

### CODE REVIEW

**Comment by dailidong** [comment]
I only check the io.jsonwebtoken jar license can use, its license is apache 2.0.

who can review this pr?

**Comment by elonlo** [comment]
Currently, I have added jwt verification of SHA and RSA algorithms and provided complete test cases. The following is a brief description of the three verification methods.
- PASSWORD：Is a mysql-based authentication method, using the username and password to retrieve the user and generate a session
- JWT_KEY：Using HMAC-SHA algorithm, the user provides a Secret Key to symmetrically encrypt user information to generate jwt
- JWT_RSA: Using RSA algorithm, support user local private key file to generation jwt or remote server to generate jwt
Please continue to help review

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/security/JsonWebTokenRSAAuthenticator.java [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+                result.setCode(Status.CREATE_USER_ERROR.getCode());
+                result.setMsg(Status.CREATE_USER_ERROR.getMsg());
+                return result;
+            }
+            result.setData(Collections.singletonMap(Constants.USER_AUTH, tokenResult.getData()));
+            result.setCode(Status.SUCCESS.getCode());
+            result.setMsg(Status.LOGIN_SUCCESS.getMsg());
+        } catch (Exception e) {
+            logger.error(e.toString());
+            e.printStackTrace();
```
Comment (Technoboy-) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/security/JsonWebTokenRSAAuthenticator.java]: delete this line

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/security/JsonWebTokenRSAAuthenticator.java [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+        claims.put(requiredNameClaim, user.getUserName());
+        claims.put(requiredEmailClaim, user.getEmail());
+        String token = Jwts.builder()
+                .setIssuedAt(Date.from(now))
+                .setExpiration(Date.from(now.plus(expireTimeOfMinutes, ChronoUnit.MINUTES)))
+                .addClaims(claims)
+                .setId(UUID.randomUUID().toString())
+                .signWith(privateKey)
+                .compact();
+        if (logger.isDebugEnabled()) {
```
Comment (Technoboy-) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/security/JsonWebTokenRSAAuthenticator.java]: check if the token is blank

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/security/AuthenticationType.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+package org.apache.dolphinscheduler.api.security;
+
+import com.baomidou.mybatisplus.annotation.EnumValue;
+
+/**
+ * authentication type
+ */
+public enum AuthenticationType {
+
+    PASSWORD(0, "verifying via user name and password"),
```
Comment (Technoboy-) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/security/AuthenticationType.java]: verify, not verifying

Comment (elonlo) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/security/AuthenticationType.java]: hmm, i've corrected it

**Pull request closed without merging**